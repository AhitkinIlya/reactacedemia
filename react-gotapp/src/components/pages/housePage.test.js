import React from 'react';
import HousePage from './housePage';
import {shallow} from 'enzyme'

describe('Testing <HousePage/>', () => {
    const housePage = shallow(<HousePage/>)
    describe('Testing state, props and snap', () => {
        it('Header have rendered correctly', () => {
            expect(housePage).toMatchSnapshot()
        })

        it('testing state "error" is false', () => {
            expect(housePage.state().error).toBeFalsy()
        })

        it('testing state "selectedItem" is number', () => {
            expect(housePage.state().selectedItem).toBe(4)
        })
    })
    describe('Testing handlers', () => {
        it('Testing componentDidCauth', () => {
            housePage.instance().componentDidCatch()
            expect(housePage.state().error).toBeTruthy()
        })

        it('Testing onItemSelected', () => {
            housePage.instance().onItemSelected(3)
            expect(housePage.state().selectedItem).toBeNumber()
        })
    })
})