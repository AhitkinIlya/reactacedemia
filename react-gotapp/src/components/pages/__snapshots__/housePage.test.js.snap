// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <HousePage/> Testing state, props and snap Header have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <housePage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "left": <ItemList
        getData={[Function]}
        onItemSelected={[Function]}
        renderItem={[Function]}
      />,
      "right": <CharDetails
        getData={[Function]}
        itemId={4}
        title="Выберите, пожалуйста, дом из списка"
      >
        <Field
          field="region"
          label="Region"
        />
        <Field
          field="coatOfArms"
          label="coatOfArms"
        />
      </CharDetails>,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "left": <ItemList
          getData={[Function]}
          onItemSelected={[Function]}
          renderItem={[Function]}
        />,
        "right": <CharDetails
          getData={[Function]}
          itemId={4}
          title="Выберите, пожалуйста, дом из списка"
        >
          <Field
            field="region"
            label="Region"
          />
          <Field
            field="coatOfArms"
            label="coatOfArms"
          />
        </CharDetails>,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
